src/models/preferences.js

```diff
@@ -82,16 +80,18 @@ class Preferences {
         }
 
         fs.stat(this._filepath, (statsErr, stats) => {
-          if (statsErr) {
-            return reject(statsErr);
-          }
-
           let updatePreferences = (preferences) => {
             preferences[key] = value;
 
-            fs.writeFileSync(this._filepath, JSON.stringify(preferences));
+            fs.writeFile(this._filepath, JSON.stringify(preferences),
+              (writeErr) => {
+                if (writeErr) {
+                  return reject(writeErr);
```

Comment:
Let's just throw here rather than reject. It's already an error, so we can propagate it as such imo:

```diff
- return reject(writeErr);
+ throw(writeErr);
```